plugins/destination/snowflake/client/read.go

```diff
@@ -12,61 +13,85 @@ const (
 	readSQL = "SELECT * FROM %s WHERE \"_cq_source_name\" = ?"
 )
 
-func (*Client) createResultsArray(table *schema.Table) []any {
+// https://github.com/snowflakedb/gosnowflake/issues/674
+func snowflakeStrToIntArray(val string) []string {
+	val = strings.TrimPrefix(val, "[\n  ")
+	val = strings.TrimSuffix(val, "\n]")
+	strs := strings.Split(val, ",\n  ")
+	for i := range strs {
+		strs[i] = strings.ReplaceAll(strs[i], "\\\"", "\"")
+		strs[i] = strings.ReplaceAll(strs[i], "\\n", "\n")
+	}
+	return strs
+}
+
+// https://github.com/snowflakedb/gosnowflake/issues/674
+func snowflakeStrToArray(val string) []string {
+	val = strings.TrimPrefix(val, "[\n  \"")
+	val = strings.TrimSuffix(val, "\"\n]")
+	strs := strings.Split(val, "\",\n  \"")
+	for i := range strs {
+		strs[i] = strings.ReplaceAll(strs[i], "\\\"", "\"")
+		strs[i] = strings.ReplaceAll(strs[i], "\\n", "\n")
+	}
+	return strs
+}
+
+func (*Client) createResultsArray(values []any, table *schema.Table) []any {
 	results := make([]any, 0, len(table.Columns))
-	for _, col := range table.Columns {
+	for i, col := range table.Columns {
 		switch col.Type {
 		case schema.TypeBool:
-			var r bool
-			results = append(results, &r)
+			r := (*values[i].(*interface{})).(bool)
+			results = append(results, r)
 		case schema.TypeInt:
-			var r int
-			results = append(results, &r)
+			r := (*values[i].(*interface{})).(string)
```

Comment:
It can be a good improvement but we are not going to implement generics right now as it doesn't have an impact on users and the value is too small yet as the codebase that involves that is still small. So going to postpone this further down the road.